library/src/main/java/com/bumptech/glide/request/SingleRequest.java

```diff
@@ -409,13 +411,20 @@ private Drawable getPlaceholderDrawable() {
   private Drawable getFallbackDrawable() {
     if (fallbackDrawable == null) {
       fallbackDrawable = requestOptions.getFallbackDrawable();
-      if (fallbackDrawable == null && requestOptions.getFallbackId() > 0) {
+      if (fallbackDrawable == null && isValidId(requestOptions.getFallbackId())) {
         fallbackDrawable = loadDrawable(requestOptions.getFallbackId());
       }
     }
     return fallbackDrawable;
   }
 
+  /**
+   * @see android.content.res.ResourceId.isValid
+   */
```

Comment:
I'd prefer not to rely on this hidden framework method to identify the validity of this ID - I'm guessing it's not exposed for a reason. This may not be consistent across SDKs and it relies on the behavior of aapt.

I think it would be better to expose `hasErrorId`, `hasPlaceholderId`, `hasFallbackId` on requestoptions that checks the underlying `fields` field instead, and make the getters return `null`.

Reply:
I agree with your point about not relying on this hidden framework method. 

However, I’d like to ask about the intention behind the code `requestOptions.getErrorId() > 0`. 
I thought this code was intended to verify whether the resource ID is valid or not.

However, based on my understanding of your suggestion, the code should be modified as follows. 

```java
if (errorDrawable == null && requestOptions.getErrorId() != null) {
        errorDrawable = loadDrawable(requestOptions.getErrorId());
      }
```

And it seems that it is not verifying the validity of the resourceID, but simply checking for its existence.
Could you clarify the purpose of this code `requestOptions.getErrorId() > 0`?

Reply:
I searched for a definitive method to validate the resourceId, but I couldn't find a clear and publicly documented way to do so. Therefore, I believe that the responsibility for validating resourceId should lie with the users themselves, as shown in the code below, rather than relying solely on the Glide library.

```kotlin
try {
    Drawable drawable = getResources().getDrawable(R.drawable.my_image);
    // Valid resource
} catch (Resources.NotFoundException e) {
    // Invalid resource
}
```

Therefore, I refactored my commit to check the existance of `errorId`, `placeholderId`, `fallbackId` as you suggested.
Please, verify it @kanelbulle

Reply:
Thanks! I looked back in the history and this check was added in https://github.com/bumptech/glide/commit/33b6412185eb0d6860c2eebbc330d5c31c4429da#diff-ee7b45dc91d122ca9ab77bb479e9daaebdf68d3705a77a09a0a5e71cba590528 combined with a number of other changes. I suspect it was an easy way to check if an id was set or not, which I think your change now better accomplishes.

Reply:
@kanelbulle
It seems that auto-merge has been disabled after I rebased my commits. 
Could you kindly take a look when you have a moment?